### PR TITLE
Introduce db migrations

### PR DESCRIPTION
**Pull request by ataleksandrov**
# Motivation
On start up database should be configured with migration schema.

# Approach
Introduce [migrate dependency](github.com/golang-migrate/migrate) for database migrations on bootstrap of go-tin.

### CODE REVIEW

File: storage/migrations/001_initial_schema.up.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+BEGIN;
```
Comment (DimitarPetrov) [inline storage/migrations/001_initial_schema.up.sql:1]: Can you unify formatting. Maybe use 4 spaces?

File: storage/migrations/001_initial_schema.up.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+CREATE TABLE users (
+  id SERIAL PRIMARY KEY NOT NULL,
+  username varchar(255) NOT NULL,
+  password varchar(255) NOT NULL,
+  birth_date date NOT NULL,
```
Comment (DimitarPetrov) [inline storage/migrations/001_initial_schema.up.sql]: Maybe birth_date, bio and hometown could be allowed to be null, so users could decide if they want this information for them to be displayed.

File: storage/migrations/001_initial_schema.up.sql [hunk 3]
```diff
@@ -0,0 +1,28 @@
+BEGIN;
+
+CREATE TABLE users (
+  id SERIAL PRIMARY KEY NOT NULL,
+  username varchar(255) NOT NULL,
```
Comment (DimitarPetrov) [inline storage/migrations/001_initial_schema.up.sql]: Maybe inline unique constraint

File: storage/migrations/001_initial_schema.up.sql [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  id SERIAL PRIMARY KEY NOT NULL,
+  title varchar(255) NOT NULL UNIQUE,
+  content text NOT NULL,
+  date date NOT NULL,
+  user_id INTEGER NOT NULL,
+  CONSTRAINT posts_users_user_id_fkey FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+);
+
+CREATE TABLE followers(
+ follower_id INTEGER NOT NULL,
```
Comment (DimitarPetrov) [inline storage/migrations/001_initial_schema.up.sql]: follower_id and target_id should be foreign keys and could be cascade deleted when user is deleted.
IMO when user is deleted it is useful all records for the users that he/she follows should be deleted too.
Also when user is deleted it must disappear from his followers following lists.

File: storage/storage.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
+package storage
+
+import (
```
Comment (DimitarPetrov) [inline storage/storage.go:3]: run go-imports to structure import statements

File: storage/storage.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+	migrationsURL := fmt.Sprintf("file://%s/migrations", basepath)
+	m, err := migrate.NewWithDatabaseInstance(migrationsURL, dbType, driver)
+	if err != nil {
+		return err
+	}
+	err = m.Up()
+	if err == migrate.ErrNoChange {
+		log.Println("No changes to the database schema have been made")
+		err = nil
```
Comment (DimitarPetrov) [inline storage/storage.go]: Directly return nil

File: storage/storage.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("unable to open db connection: %s", err)
+	}
+
+	err = updateSchema(db, "postgres")
+	if err != nil {
+		return nil, fmt.Errorf("unable to update db schema: %s", err)
+	}
+
+	log.Println("Database is up-to-date")
```
Comment (DimitarPetrov) [inline storage/storage.go:97]: Use info level of logging

**Review by DimitarPetrov — CHANGES_REQUESTED** [review]

File: storage/storage.go (hunk 7 above)
Comment (ataleksandrov) [reply storage/storage.go:97]: Not applicable for log build in package.

**Review by DimitarPetrov — APPROVED** [review]

**Pull request merged**